numcodecs/tests/test_zarr3.py

```diff
@@ -0,0 +1,206 @@
+from __future__ import annotations
+from typing import Iterator
+
+
+import numpy as np
+import pytest
+import sys
+
+from numcodecs.registry import get_codec
+
+try:
+    from zarr.codecs.registry import get_codec_class
+    from zarr.array import Array
+    from zarr.common import JSON
+    from zarr.codecs import BytesCodec
+    from zarr.abc.store import Store
+    from zarr.store import MemoryStore, StorePath
+
+except ImportError:
+    pass
+
+
+pytestmark = pytest.mark.skipif(
+    sys.version_info < (3, 10), reason="zarr-python 3 requires Python 3.10 or higher"
+)
+
+
+@pytest.fixture
+def store() -> Iterator[Store]:
+    yield StorePath(MemoryStore(mode="w"))
+
+
+@pytest.mark.parametrize(
+    "codec_id", ["blosc", "lz4", "zstd", "zlib", "gzip", "bz2", "lzma", "shuffle"]
+)
+def test_generic_codec(store: Store, codec_id: str):
+    data = np.arange(0, 256, dtype="uint16").reshape((16, 16))
+
+    with pytest.warns(UserWarning, match="Numcodecs.*"):
```

Comment:
Can you make this match more specific? It's not clear what is's warning on just from Numcodecs.*